apps/verticalhome/test/marionette/grid_layout_test.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+var assert = require('assert');
+
+marionette('Vertical - grid', function() {
+
+  var client = marionette.client(require(__dirname + '/client_options.js'));
+  var actions, home, system;
+
+  setup(function() {
+    actions = client.loader.getActions();
+    home = client.loader.getAppClass('verticalhome');
+    system = client.loader.getAppClass('system');
+    system.waitForStartup();
+    home.waitForLaunch();
+  });
+
+  test('apps grid is LTR in LTR language', function() {
+    var communicationManifestUrl =
+        'app://communications.gaiamobile.org/manifest.webapp';
+    var phoneIcon = home.getIcon(communicationManifestUrl, 'dialer').location();
+    var contactsIcon = home.getIcon(communicationManifestUrl, 'contacts')
+      .location();
+    var messageIcon = home.getIcon('app://sms.gaiamobile.org/manifest.webapp')
+      .location();
+
+    // check that they are all on the same line
+    assert.equal(phoneIcon.y, messageIcon.y,
+                 'Phone and message icons should be on the same line.');
+    assert.equal(messageIcon.y, contactsIcon.y,
+                 'Message and contacts icon should be on the same line.');
+    // check the ordering
+    assert.ok(phoneIcon.x < messageIcon.x,
+              'Message icon should be on the right of phone icon');
+    assert.ok(messageIcon.x < contactsIcon.x,
+              'Contacts icon should be on the right of phone icon');
+  });
+
+  test('apps grid is RTL in RTL language', function() {
+    var communicationManifestUrl =
+        'app://communications.gaiamobile.org/manifest.webapp';
+
+    client.executeScript(function() {
+      navigator.mozSettings.createLock().set({
+        'language.current': 'ar'
+      });
+    });
+    // Localization can be async, wait for the content to update
+    client.waitFor(function() {
+      var phoneIcon = home.getIcon(communicationManifestUrl, 'dialer');
+      return phoneIcon.text() ===
+        home.localizedAppName('communications', 'dialer', 'ar');
+    });
+
+    var phoneIcon = home.getIcon(communicationManifestUrl, 'dialer').location();
+    var contactsIcon = home.getIcon(communicationManifestUrl, 'contacts')
+      .location();
+    var messageIcon = home.getIcon('app://sms.gaiamobile.org/manifest.webapp')
+      .location();
```

Comment:
Same here, just get the first two icons and test that the first is aligned to the right and the second is to the left of the first.